internal/build/buildconfig/manager.go

```diff
@@ -34,6 +37,11 @@ func NewManager(client client.Client, maker Maker, ocpBuildsHelper OpenShiftBuil
 func (bcm *buildConfigManager) Sync(ctx context.Context, mod kmmv1beta1.Module, m kmmv1beta1.KernelMapping, targetKernel string, pushImage bool) (build.Result, error) {
 	logger := log.FromContext(ctx)
 
+	buildConfigTemplate, err := bcm.maker.MakeBuildConfigTemplate(mod, m, targetKernel, m.ContainerImage, pushImage)
+	if err != nil {
+		return build.Result{}, fmt.Errorf("could not make BuildConfig template: %v", err)
```

Comment:
```suggestion
		return build.Result{}, fmt.Errorf("could not make BuildConfig template: %w", err)
```

Reply:
If i remember correctly, we agreed that we will be using %v

Reply:
Did we? I remember we said we want to wrap errors (using %w). Also I see we are using it extensively:
```
root kernel-module-management-midstream (main) $ git grep %w | grep -v vendor | grep %w
controllers/module_reconciler.go:               return res, fmt.Errorf("failed to get the requested %s KMMO CR: %w", req.NamespacedName, err)
controllers/module_reconciler.go:               return res, fmt.Errorf("could get targeted nodes for module %s: %w", mod.Name, err)
controllers/module_reconciler.go:               return res, fmt.Errorf("could get kernel mappings and nodes for modules %s: %w", mod.Name, err)
controllers/module_reconciler.go:                       return res, fmt.Errorf("failed to handle build for kernel version %s: %w", kernelVersion, err)
controllers/module_reconciler.go:               return res, fmt.Errorf("could handle device plugin: %w", err)
controllers/module_reconciler.go:               return res, fmt.Errorf("failed to update status of the module: %w", err)
controllers/module_reconciler.go:               return false, fmt.Errorf("failed to check image existence for kernel %s: %w", kernelVersion, err)
controllers/module_reconciler.go:               return false, fmt.Errorf("could not synchronize the build: %w", err)
controllers/module_reconciler.go:               return fmt.Errorf("failed to get the device plugin daemonset %s/%s: %w", name, mod.Namespace, err)
controllers/module_reconciler.go:               return nil, fmt.Errorf("failed to get the kmmo module %s: %w", namespacedName, err)
internal/auth/auth.go:          return nil, fmt.Errorf("cannot find secret %s: %w", rsag.namespacedName, err)
internal/auth/auth.go:          return nil, fmt.Errorf("could not create a keycahin from secret %v: %w", secret, err)
internal/daemonset/daemonset.go:                return nil, fmt.Errorf("could not get all DaemonSets: %w", err)
internal/module/kernelmapper.go:                return nil, fmt.Errorf("failed to substitute the os config into ContainerImage field: %w", err)
internal/registry/registry.go:          return false, fmt.Errorf("failed to get pull options for image %s: %w", image, err)
internal/registry/registry.go:          return false, fmt.Errorf("could not get image %s: %w", image, err)
internal/registry/registry.go:          return nil, nil, fmt.Errorf("failed to get pull options for image %s: %w", image, err)
internal/registry/registry.go:          return nil, nil, fmt.Errorf("failed to get manifest from image %s: %w", image, err)
internal/registry/registry.go:          return nil, nil, fmt.Errorf("failed to get layers digests from manifest of the image %s: %w", image, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to find file %s in image layer: %w", driverToolkitJSONFile, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get KERNEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get RT_KERNEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get RHEL_VERSION from %s, found %t: %w", driverToolkitJSONFile, found, err)
internal/registry/registry.go:                  return nil, fmt.Errorf("cannot get keychain from the registry auth getter: %w", err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get manifest stream from image %s: %w", image, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get crane manifest from image %s: %w", image, err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to unmarshal crane manifest: %w", err)
internal/registry/registry.go:          return nil, fmt.Errorf("unmarshalled manifests invalid format: %w", err)
internal/registry/registry.go:                  return nil, fmt.Errorf("failed to get arch digets from multi arch image: %w", err)
internal/registry/registry.go:                  return nil, fmt.Errorf("failed to get crane manifest for the arch image: %w", err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to unmarshal manifest stream: %w", err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to get targz from layer: %w", err)
internal/registry/registry.go:          return nil, fmt.Errorf("failed to create reader from targz: %w", err)
internal/registry/registry.go:                  return nil, fmt.Errorf("failed to get next entry from targz: %w", err)
internal/registry/registry.go:                          return nil, fmt.Errorf("failed to read tar entry: %w", err)
internal/registry/registry.go:                          return nil, fmt.Errorf("failed to unmarshal tar entry: %w", err)
internal/registry/registry.go:          return "", fmt.Errorf("failed to unmarshal manifest stream: %w", err)
```

Reply:
we are also using %v extensivily